backend/Application/Aggregates/Contract/Resilience/Policies.cs

```diff
@@ -0,0 +1,76 @@
+using System.Threading.Tasks;
+using Npgsql;
+using Polly;
+
+namespace Application.Aggregates.Contract.Resilience;
+
+internal static class Policies
+{
+    /// <summary>
+    /// Create a async retry policy which retries on all transient database errors.
+    /// </summary>
+    internal static AsyncPolicy GetTransientPolicy(ILogger logger, int retryCount, TimeSpan delay)
+    {
+        var policyBuilder = Policy
+            .Handle<NpgsqlException>(ex => ex.IsTransient)
+            .OrInner<NpgsqlException>(ex => ex.IsTransient);
+        AsyncPolicy policy;
+        if (retryCount == -1)
+        {
+            policy = policyBuilder
+                .WaitAndRetryForeverAsync((_, _, _) => delay,
+                    (ex, currentRetryCount, _, _) =>
+                    {
+                        logger.Error(ex, $"Triggering retry policy with {currentRetryCount} due to exception");
+                        return Task.CompletedTask;
+                    });
+        }
+        else
+        {
+            policy = policyBuilder
+                .WaitAndRetryAsync(retryCount,
+                    (_, _, _) => delay,
+                    (ex, _, currentRetryCount, _) =>
+                    {
+                        logger.Error(ex, $"Triggering retry policy with {currentRetryCount} due to exception");
+                        return Task.CompletedTask;
+                    });
+        }
+
+        return policy;
+    }    
+    
+    /// <summary>
+    /// Create a async retry policy which retries on all transient database errors.
+    /// </summary>
+    internal static AsyncPolicy<T> GetTransientPolicy<T>(ILogger logger, int retryCount, TimeSpan delay)
```

Comment:
Why do you need to copy the entire function here for a "Generic version"? :thinking: 
( / Or why do you need the non-generic)

Reply:
This is because of the API’s used in the framework. It has two difference builders dependent if there is a return type or not.

Reply:
Okay, it just seems weird that you have to implement the (seemingly) same function twice :shrug: